hera_cal/lstbin.py

```diff
@@ -552,70 +547,52 @@ def config_lst_bin_files(data_files, dlst=None, atol=1e-10, lst_start=None, lst_
     time_arrays = []
     for di, dfs in enumerate(data_files):
         # get times
-        dlsts, dtimes, larrs, tarrs = io.get_file_times(dfs, filetype='uvh5')
-
-        # get lmin: LST of first integration from first file
-        if di == 0:
-            lmin = larrs[0][0]
-
-        # unwrap relative to lmin
-        for la in larrs:
-            if la[0] < lmin:
-                la += 2 * np.pi
-
-        # get lmax
-        if di == (len(data_files) - 1):
-            lmax = larrs[-1][-1]
-
+        _, _, larrs, tarrs = io.get_file_times(dfs, filetype='uvh5')
         # append
         lst_arrays.append(larrs)
         time_arrays.append(tarrs)
 
-    # get starting LST for output binning
+    # get begin_lst from lst_start or from the first JD in the data_files
     if lst_start is None:
-        lst_start = lmin
-
-    # if lst_start is sufficiently below lmin, shift everything down an octave
-    if lst_start < (lmin - np.pi):
-        lst_arrays = [[la - 2 * np.pi for la in day] for day in lst_arrays]
-        lmin -= 2 * np.pi
-        lmax -= 2 * np.pi
-
-    # get beginning LST for lst_grid
-    if fixed_lst_start:
-        begin_lst = lst_start
-    else:
-        begin_lst = lmin
-
-    # get stopping LST for output binning
-    if lst_stop is None:
-        lst_stop = lmax
-    if lst_stop < begin_lst:
-        lst_stop += 2 * np.pi
+        all_lsts = [lst for larrs in lst_arrays for larr in larrs for lst in larr]
+        all_times = [time for tarrs in time_arrays for tarr in tarrs for time in tarr]
+        lst_start = all_lsts[np.argmin(all_times)]
```

Comment:
why argmin all_times?

Reply:
Not at all. This is simply asserting that the starting LST should correspond to the first JD on the first night. This method allows the days to come in in any order and uses their JDs (which don't have a phase wrap, unlike lst) to figure out which LST is "first."

Reply:
Consider the following example: let's say I want to LST bin two (or more) nights from either end of the year (or perhaps more realistically, across different seasons). Let's say these are JDs 2458100.2-.6 and 2458400.2-.6 (almost a year separated). These corresponds to JDs of 6.19 - 2.43 radians and 5.07 - 1.31 radians, respectively. This argmin will grad 2458100.2 as the starting LST, so the LST grid will be 6.19 - 12.47 radians, but there is data before that 6.19 radians coming from the second day, which will then get reflected up to 11.3 radians. This could not be a problem, because those LSTs will be binned and then re-reflected down to 5.07 radians, but I wonder if these files may actually just be dropped? Or will this not happen?

Reply:
also this may pose an issue for binning the first (and last) bin of the grid. before we used lst_start and lst_end to specify an LST grid that spans across and beyond the data entirely (assuming we were binning data that did not span exactly 24 hours). with a odd reflection like what I describe above, its possible that the first and last LST bin of the grid will not grab the data that would have fallen in the bin on the other side, if you know what I mean

Reply:
Regarding the first issue, of trying to bin together two days from different seasons, this edge case can be easily addressed by manually providing an `lst_start`. I'm trying to do the most sensible thing where no `lst_start` is provided and I think this is it. The only alternative is making the `lst_start` parameter mandatory, but I feel like that's more API-breaking than this current PR. 

The current code that drops files has logic borrowed by LST-bin files that makes sure that no data gets dropped. I'm not seeing how we'd lose an integration.

Reply:
right I forgot that lst_start is still a parameter, okay sounds good thanks!